plugin/advisor/catalog/finder.go

```diff
@@ -8,8 +8,8 @@ type Finder struct {
 }
 
 // NewFinder creates a new finder.
-func NewFinder(database *Database) *Finder {
-	return &Finder{database: newDatabaseState(database)}
+func NewFinder(database *Database, context *Context) *Finder {
```

Comment:
I am wondering if we could implement Finder as a Catalog interface.

func NewFinder(database *Database) *Finder {
  return &Finder{database: newDatabaseState(database)}
}

This is called where we have the catalog info, e.g https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/server/sql.go?L206-213&subtree=true 

func NewFinder() *Finder {
  return &Finder{database: emptyDatabaseState()}
}

This is called where we do not have the catalog info, e.g https://sourcegraph.com/github.com/bytebase/bytebase/-/blob/sql-server/advise.go?L105&subtree=true

It's inside databaseState new method, we determine whether we should turn on the CheckCatalog bit